### DISCUSSION TITLE
Interpreting elasticities in context of cooperative trade-offs

### DISCUSSION THREAD

**Discussion by vrmarcelino** [micom-dev/micom#50] [Q&A] [answered]
Hi!

It has been a pleasure to play with MICOM in the last few months! I am now trying to understand elasticities

Using this result as an example:

  | reaction | taxon | effector | direction | elasticity | type
-- | -- | -- | -- | -- | -- | --
0 | EX_gly_e_m | medium | S8C2322 | forward | -9.49 | abundance
1 | EX_gly_e | C_e__S11C382 | S8C2322 | forward | -0.95 | abundance
2 | EX_gly_e | C_e__S19C1549 | S8C2322 | forward | 1.71 | abundance
3 | EX_gly_e | C_e__S17C241 | S8C2322 | reverse | 2.41 | abundance

I have calculated the effect of adding species ‘S8C2322’ (potential probiotic) on Glycine production/consumption as the sum of the elasticity coefficients (only the ones in forward direction, as I think reverse direction would indicate removing this species from the host, is that correct?).

So I predicted that species ‘S8C2322’ will increase the exchange of Glycine among microorganisms by 0.76 (-0.95 + 1.71) units, but it will consume from the medium 9.49 units, right?

I have also calculated the cooperative trade-offs, and found that the net exchange fluxes for Glycine were, say, 0.5 mmol/[gDW h].

If I want to estimate net predicted change of Glycine after adding my probiotic, would you recommend multiplying  0.5 x 0.95 (and I understand that the units would be meaningless), or the net elasticity values as calculated above are the best representation of this probiotic intervention?

Also - sometimes the elasticity coefficients seem very large (>4000). Does that suggest that something went wrong?

Thanks a lot for developing this tool!

**Comment by cdiener** [comment, marked as answer]
Hi, yeah you are very close in your interpretation. So the direction is what tells you the sign of the flux and as you stated correctly it will tell you if something is imported or exported. Usually, reverse would be import, and forward would be export since those are written as `A -> ` in most models (depends a bit on your import models though). So I would guess all the forward ones are exports and the last reverse one is import. The elasticity tells you how the flux changes when the effector is increased. Negative means flux is decreased and positive means flux is increased. So my interpretation of those results would be that overall glycine production is going up (more excess glycine in EX_gly_e_m), that S11C382 produces a little less glycine, S19C1549 a little more and that S17C241 consumes more glycine. 
For overall production I would probably build two models, one without S8C2322, and one with S8C2322 and quantify the glycine production flux (sum v_i * a_i).  If you use the grow workflow and get the GrowthResults from there you could get the production rates as 

```
res = grow(...)
exports = res.exchanges[(exchanges.taxon != "medium") & (exchanges.direction == "export")]
production = exports.groupby(["sample_id", "metabolite"]).apply(lambda df: sum(df.abundance * df.flux.abs()).reset_index()
```

`production` then contains the overall production fluxes for each metabolite and sample. I definitely plan to add some helper functions to calculate production and consumption fluxes.

Very large elasticities can happen, either because of large stoichiometric or some non-uniqueness in the solution.

**Comment by vrmarcelino** [comment]
Hi!

Thanks a lot for the quick and detailed response! It is slowly sinking in :)

I'll give the it a try!
